src/library/dlgtrackinfo.cpp

```diff
@@ -296,12 +300,27 @@ void DlgTrackInfo::reloadTrackBeats(const Track& track) {
     bpmTap->setEnabled(!m_trackHasBeatMap);  // when we have a beatmap
 
     if (track.isBpmLocked()) {
-        tabBPM->setEnabled(false);
+        enableBPMControls(false);
```

Comment:
Updating the check box state should invoke the corresponding slot. Calling enableBPMControls() should be redundant here.

```
bpmLock->setChecked(track.isBpmLocked());
```